public/app/core/components/TagFilter/TagFilter.tsx

```diff
@@ -45,29 +49,70 @@ export const TagFilter: FC<Props> = ({
   tags,
   width,
 }) => {
-  const theme = useTheme();
-  const styles = getStyles(theme);
+  const styles = useStyles2(getStyles);
 
-  const onLoadOptions = async (query: string) => {
+  const currentlySelectedTags = tags.map((tag) => ({ value: tag, label: tag, count: 0 }));
+  const [options, setOptions] = useState<TagSelectOption[]>(currentlySelectedTags);
+  const [isLoading, setIsLoading] = useState(false);
+  const [previousTags, setPreviousTags] = useState(tags);
+
+  // Necessary to force re-render to keep tag options up to date / relevant
+  const selectKey = useMemo(() => tags.join(), [tags]);
+
+  const onLoadOptions = useCallback(async () => {
     const options = await tagOptions();
-    return options.map((option) => ({
-      value: option.term,
-      label: option.term,
-      count: option.count,
-    }));
-  };
+    return options.map((option) => {
+      if (tags.includes(option.term)) {
+        return {
+          value: option.term,
+          label: option.term,
+          count: 0,
+        };
+      } else {
+        return {
+          value: option.term,
+          label: option.term,
+          count: option.count,
+        };
+      }
+    });
+  }, [tagOptions, tags]);
 
-  const debouncedLoadOptions = debounce(onLoadOptions, 300);
+  const onFocus = useCallback(async () => {
+    setIsLoading(true);
+    const results = await onLoadOptions();
+    setOptions(results);
+    setIsLoading(false);
+  }, [onLoadOptions]);
+
+  useEffect(() => {
+    // Load options when tag is selected externally
+    if (tags.length > 0 && options.length === 0) {
+      onFocus();
+    }
+  }, [onFocus, options.length, tags.length]);
+
+  useEffect(() => {
+    // Update selected tags to not include (counts) when selected externally
+    if (tags !== previousTags) {
+      setPreviousTags(tags);
+      onFocus();
+    }
+  }, [onFocus, previousTags, tags]);
 
```

Comment:
Not super excited about these useEffects, but they are necessary to load options when user selects a tag externally (i.e. in table) / to ensure that selected tags externally don't have (counts) next to them (which get stale)